src/js/apps/patients/patient/sidebar/sidebar_app.js

```diff
@@ -11,32 +12,36 @@ export default App.extend({
     this.orgEngagementEnabled = currentOrg.getSetting('engagement');
     this.patient = patient;
 
-    this.showView(new SidebarView({ 
+    const widgets = this.getSidebarWidgets();
+
+    this.showView(new SidebarView({
       model: this.patient,
-      orgEngagementEnabled: this.orgEngagementEnabled,
+      collection: widgets,
     }));
-    
-    if (this.orgEngagementEnabled) {
-      this.showChildView('engagement', new EngagementStatusPreloadView());
-    }
   },
   beforeStart({ patient }) {
     return Radio.request('entities', 'fetch:patient:engagementStatus', patient.id);
   },
-  onStart() {
-    this.showEngagementStatus();
-  },
   onFail() {
-    this.showEngagementStatus();
+    this.patient.set('engagement', {
```

Comment:
At the moment, `engagement` doesn't exist on the patient until the engagement status request succeeds. When the request fails, without this line, it never gets set, so we're unable to differentiate between a loading state and a not available state in the view. And since we're now handling the preload logic in the view rather than in the app, listening for model change events seems to be the cleanest method of handling the preload logic.

I like the `_statusNotAvailable` approach. Are you suggesting that be applied directly to the patient, or nested under `engagement`

Reply:
ah so maybe just `this.patient.trigger('status:notavailable')` would be better... then the view can just set a flag on itself.  The patient model will live on in memory so it's awkward to store that state there when it may just be a result of a fluke error.